### PR TITLE
Allow usage of a typedef type in parameters of a parser for p4c-bm2-ss

### PR DESCRIPTION
**Pull request by VolodymyrPeschanenkoIntel**
If we try to get a type of a header from `refMap` for `Type_Name` then It could be represented as `Type_Typedef` type which contains type of a header in one of it's filed.
Please, run the following command to demonstrate the problem:
`./p4c-bm2-ss -o ./tmp/strength-simple.p4.json ../testdata/p4_16_samples/strength-simple.p4`

### CODE REVIEW

**Review by mbudiu-vmw — COMMENTED** [review]
Don't we have a pass to eliminate all typedefs from the program?

**Comment by VolodymyrPeschanenkoIntel** [comment]
> Don't we have a pass to eliminate all typedefs from the program?

In that case behavioral-model crashed for few examples.
For example:
1550/2217 Test  #449: bmv2/testdata/p4_16_samples/ipv6-switch-ml-bmv2.p4 .............................................***Failed    0.85 sec
Error compiling
/p4c/testdata/p4_16_samples/ml-headers.p4(182): [--Werror=type-error] error: Field _egress_port1 is not a member of structure struct metadata_t
struct metadata_t {
       ^^^^^^^^^^
/p4c/testdata/p4_16_samples/ml-headers.p4(182): [--Werror=type-error] error: Field _fwd_out_bd69 is not a member of structure struct metadata_t
struct metadata_t {
       ^^^^^^^^^^
/p4c/testdata/p4_16_samples/ml-headers.p4(182): [--Werror=type-error] error: Field _fwd_out_bd69 is not a member of structure struct metadata_t
struct metadata_t {
       ^^^^^^^^^^

*** Test failed

But everything is correct in p4c side.

File: backends/bmv2/CMakeLists.txt [hunk 1]
```diff
@@ -181,6 +181,8 @@ set (XFAIL_TESTS
   testdata/p4_16_samples/issue1882-bmv2.p4
   testdata/p4_16_samples/issue1882-1-bmv2.p4
   testdata/p4_16_samples/issue2664-bmv2.p4
+  # typedef type as a parameters of a parser
+  testdata/p4_16_samples/strength-simple.p4
```
Comment (mbudiu-vmw) [inline backends/bmv2/CMakeLists.txt:185]: why is this test failing?

**Comment by mbudiu-vmw** [comment]
Doesn't #3123 solve this problem?
If so, please approve that PR and close this one.

**Comment by VolodymyrPeschanenkoLitSoft** [comment]
Implemented in #3123

**Pull request closed without merging**